runtime/src/bank.rs

```diff
@@ -1211,8 +1204,37 @@ impl Bank {
     }
 
     pub fn add_native_program(&self, name: &str, program_id: &Pubkey) {
-        let account = native_loader::create_loadable_account(name);
-        self.store_account(program_id, &account);
+        let mut already_genuine_program_exists = false;
+        if let Some(mut account) = self.get_account(&program_id) {
+            already_genuine_program_exists = native_loader::check_id(&account.owner);
+
+            if !already_genuine_program_exists {
```

Comment:
Bring on the fire and brimstone.... what if this account has a very high balance and is obviously worth a lot to someone.  Seems like instead or along with we should block known addresses?

Reply:
> Bring on the fire and brimstone.... 

Hehe, I like thinking hard for corner cases... :)

> Also, what if someone adds an account and assigns it to the native_loader?

Yeah, I thought of it. Well, can someone assign accounts to non-existing owner program_id (=`NativeLoader1111111111111111111111111111111`)?:

```
ryoqun@ubuqun:~/work/solana/solana$ ./target/release/solana  --url http://api.mainnet-beta.solana.com account BPFLoader1111111111111111111111111111111111

Public Key: BPFLoader1111111111111111111111111111111111
Balance: 0.000000001 SOL
Owner: NativeLoader1111111111111111111111111111111
Executable: true
Rent Epoch: 0
Length: 25 (0x19) bytes
0000:   73 6f 6c 61  6e 61 5f 62  70 66 5f 6c  6f 61 64 65   solana_bpf_loade
0010:   72 5f 70 72  6f 67 72 61  6d                         r_program


ryoqun@ubuqun:~/work/solana/solana$ ./target/release/solana  --url http://api.mainnet-beta.solana.com account NativeLoader1111111111111111111111111111111
Error: AccountNotFound: pubkey=NativeLoader1111111111111111111111111111111

```

I just blindly thought the original code was covering the case that with mere this condition: `native_loader::check_id(&account.owner)`.

If someone can do the wild, we can add `account.executable` condition to discern legitimate native loader program from the spoofed native loader program?

Reply:
> what if this account has a very high balance and is obviously worth a lot to someone. Seems like instead or along with we should block known addresses from being added in the first place?

;) Yeah, this could be quite a draconian. But, just to be sure, is this really protecting someone from misfortune? Well, people can give lump of sols into wrong account with 1 letter typo... Also, people can send to any of sysvars...

Anyway, I think we should need a preparatory pr to prohibit money transfer to & from some pretty long prefix or suffix? Considering new sysvar or new bpf loaders are rare, this might be a too much. Also, finding nice base58-friendly prefix/postfix seems hard....

Or, I'm also thinking to practice of 2-phased introduction of sysvars/loaders like this:
1(a). Release a patch release which just prohibits any money transfer to certain account at certain epoch.
1(b). Or, just create an account with Placeholder11111 with the executable bit? (might be easiest?)
1(c). Add BPFLoader11111 to frozen account.... (no, this is bad, dos vector)
2. release the acutual meat